test/recipes/82-test_ocsp_cert_chain.t

```diff
@@ -0,0 +1,95 @@
+#! /usr/bin/env perl
+# Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+use IPC::Open3;
+use OpenSSL::Test qw/:DEFAULT srctop_file bldtop_file/;
+use OpenSSL::Test::Utils;
+use Symbol 'gensym';
+
+my $test_name = "test_ocsp_cert_chain";
+setup($test_name);
+
+plan skip_all => "$test_name requires OCSP support"
+    if disabled("ocsp");
+plan skip_all => "$test_name requires EC cryptography"
+    if disabled("ec");
+plan skip_all => "$test_name requires sock enabled"
+    if disabled("sock");
+plan skip_all => "$test_name requires TLS enabled"
+    if alldisabled(available_protocols("tls"));
+plan skip_all => "$test_name is not available Windows or VMS"
+    if $^O =~ /^(VMS|MSWin32)$/;
+
+plan tests => 2;
+
+my $shlib_wrap   = bldtop_file("util", "shlib_wrap.sh");
+my $apps_openssl = bldtop_file("apps", "openssl");
+
+my $index_txt             = srctop_file("test", "ocsp-tests", "index.txt");
+my $ocsp_pem              = srctop_file("test", "ocsp-tests", "ocsp.pem");
+my $intermediate_cert_pem = srctop_file("test", "ocsp-tests", "intermediate-cert.pem");
+
+my $server_pem            = srctop_file("test", "ocsp-tests", "server.pem");
+
+sub run_test {
+
+    # openssl ocsp -port 19254 -index index.txt -rsigner ocsp.pem -CA intermediate-cert.pem
+    my @ocsp_cmd = ("ocsp", "-port", "19254", "-index", $index_txt, "-rsigner", $ocsp_pem, "-CA", $intermediate_cert_pem);
```

Comment:
These fixed port numbers are problematic when the build is simultaneously running multiple times on the same machine. Could you please make the port number here and below a random one from some range (i.e., 19200-19299 should be good enough).

Reply:
Although it will be complicating the test quite significantly as the certificate will have to be generated on-fly.

Reply:
Alternatively we could check here for the failure of starting the ocsp responder on the fixed port and just skip the test in such case. That could be easier to do.

The s_server port could be still randomized.

Reply:
thanks for your comments, Tomas.

I will randomize the s_server port.

I agree that randomizing the ocsp port would be cumbersome.  I did include a script to regenerate the various keys and certs, but randomizing the ocsp port would seem to require updating `ca.cnf` (unless there is a way to pass that info via an option).

Reply:
> check here for the failure of starting the ocsp responder on the fixed port and just skip the test in such case.

The danger with skipping is that the port the ocsp responder wants might become unavailable permanently (e.g. maybe it is claimed by another test).  The result would be that the test stops being run and we do not notice because there is no failure message.

Reply:
We discussed the busy-port problem at the stand-up today.

Matt C made the suggestion to introduce a change to `s_server` that allows an environment variable to override the OCSP url from the cert.  We could use that to direct the OCSP query to the correct port.

Reply:
after looking at the s_server code, I saw there already exists an option `-status_url` that can be used to specify the url for the ocsp responder when s_server is run.  That url is used when 
the `authorityInfoAccess` extension is not present in the server cert.

I've updated the test to use `-status_url` and now both ports can be randomized.